File: <100/075/075.jl
function primitivePythagoreanGenerator(n :: Int, m :: Int) :: Tuple{Int, Int, Int}
  if n ≤ m || 1 < gcd(n, m) || n%2 == m%2
    return (-1, -1, -1)
  end

  (a, b, c) = (n^2 - m^2, 2 * n * m, n^2 + m^2)
  if b < a
    a, b = b, a
  end
  
  return (a, b, c)
end


function main() :: Nothing
  max_L = 1500000
  counts = [0 for _ ∈ 1:max_L]
  for n ∈ 1:max_L
    if max_L <= n^2
      break
    end
    for m ∈ 1:n
      (a, b, c) = primitivePythagoreanGenerator(n, m)
      if a < 0
        continue
      end

      L = a + b + c
      buf = L
      while L ≤ max_L
        counts[L] += 1
        L += buf
      end
    end
  end

  counts |> v -> filter(x -> x == 1, v) |> sum |> println
end


@time main()

File: <100/075/075.py
import math

def primitivePythagoreanGenerator(n: int, m: int) -> tuple[int, int, int]:
    if n <= m or math.gcd(n, m) > 1 or n%2 == m%2:
        return (-1, -1, -1)
    
    a, b, c = n**2 - m**2, 2*n*m, n**2 + m**2
    if b < a:
        a, b = b, a
    
    return (a, b, c)


def main() -> int:
    max_L = 1500000
    counts = [0 for _ in range(max_L+1)]
    for n in range(2, max_L):
        if max_L <= n*n:
            break
        for m in range(1, n):
            (a, b, c) = primitivePythagoreanGenerator(n, m)
            if a < 0:
                continue
            L = a + b + c
            buf = L
            while L <= max_L:
                counts[L] += 1
                L += buf
    
    filtered = filter(lambda x: x == 1, counts)
    return len(list(filtered))


if __name__ == "__main__":
    answer = main()
    print(answer)
